packages/library/src/providers/CFDProvider/CFDProvider.tsx

```diff
@@ -0,0 +1,45 @@
+import React, { createContext, useCallback, useMemo, useState } from 'react';
+import { TMarketTypes, TPlatforms, THooks } from '../../types';
+
+type TCFDState = {
+    // Add your CFD states here
+    marketType?: TMarketTypes.All;
+    platform?: TPlatforms.All;
+    selectedJurisdiction?: THooks.AvailableMT5Accounts['shortcode'];
+};
+
+type TCFDContext = {
+    getCFDState: <T extends keyof TCFDState>(key: T) => TCFDState[T];
+    setCfdState: <T extends keyof TCFDState>(key: T, value: TCFDState[T]) => void;
+};
+
+const CFDContext = createContext<TCFDContext | null>(null);
+
+export const useCFDContext = () => {
+    const context = React.useContext(CFDContext);
+
+    if (!context) {
+        throw new Error('useCFDContext must be used within a CFDProvider');
```

Comment:
```suggestion
        throw new Error('useCFDContext must be used within a @deriv/library/Provider.CFDProvider');
```
Makes it clear that you need to import it from the `@deriv/library` package